cmd/common/oci.go

```diff
@@ -256,12 +261,57 @@ func NewRunCommand(rootCmd *cobra.Command, runtime runtime.Runtime, hiddenColumn
 
 		timeoutDuration := time.Duration(timeoutSeconds) * time.Second
 
-		imageName := args[0]
+		var imageName string
+		if len(args) > 0 {
+			imageName = args[0]
+		}
+
+		paramValueMap := make(map[string]string)
+
+		if inFile != "" {
+			f, err := os.Open(inFile)
+			if err != nil {
+				return fmt.Errorf("opening gadget runtime configuration file %s: %w", inFile, err)
+			}
+			specs, err := gadgetconfig.InstanceSpecsFromReader(f)
+			f.Close()
+			if err != nil {
+				return fmt.Errorf("reading gadget runtime configuration file %s: %w", inFile, err)
+			}
+
+			detachedParam := runtimeParams.Get("detach")
+			isDetach := detachedParam != nil && detachedParam.AsBool()
+
+			if len(specs) > 1 && !isDetach {
+				return fmt.Errorf("multiple gadget runtime configurations found in file %s: this is only supported in combination with --detach", inFile)
+			}
+
+			if isDetach {
+				// Copy special oci params
+				ociParams.CopyToMap(paramValueMap, "operator.oci.")
+				return runInstanceSpecsDetached(ctx, runtime, specs, runtimeParams, paramValueMap,
+					gadgetcontext.WithDataOperators(ops...),
+					gadgetcontext.WithTimeout(timeoutDuration),
+					gadgetcontext.WithUseInstance(commandMode == CommandModeAttach),
+				)
+			}
+
+			cfg := specs[0]
+			imageName = cfg.ImageName
+			runtimeParams.Set("id", cfg.ID)
+			runtimeParams.Set("name", cfg.Name)
+			runtimeParams.Set("tags", strings.Join(cfg.Tags, ","))
+		}
```

Comment:
Add `paramValueMap = cfg.ParamValues`, otherwise parameters on the file will be ignored when running `... run -f foo.yaml` (without --detach)